Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -51,30 +58,33 @@ class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor<Site>, SearchViewController
             inProgress = deferred as? Cancellable
 
             deferred.upon() { result in
-                self.inProgress = nil
-
+                DispatchQueue.main.async {
+                    self.inProgress = nil
+                }
                 // Failed cursors are excluded in .get().
-                if let cursor = result.successValue {
-                    // First, see if the query matches any URLs from the user's search history.
-                    self.load(cursor)
-                    for site in cursor {
-                        if let url = site?.url,
-                               let completion = self.completionForURL(url) {
-                            DispatchQueue.main.async {
-                                self.urlBar.setAutocompleteSuggestion(completion)
-                            }
-                            return
+                guard let cursor = result.successValue else {
+                    return
```

Comment:
Don't you want to try matching one of the Alexa domains here?

Reply:
I only want to fix the main thread disk I/O in this patch.
Performance-wise I'd rather do as little changes as possible unless I spend more time on this code. I'd want to get a chunk of time allocated to do that rather than semi-drive-by patching that I am doing now.

Reply:
I mention this because I think you've changed the behavior.